src/main/scala/it/unibo/authsim/library/dsl/cryptography/encrypter/asymmetric/RSACipher.scala

```diff
@@ -0,0 +1,108 @@
+package it.unibo.authsim.library.dsl.cryptography.encrypter.asymmetric
+
+import it.unibo.authsim.library.dsl.cryptography.algorithm.AsymmetricEncryptionAlgorithm
+import it.unibo.authsim.library.dsl.cryptography.algorithm.asymmetric.RSA
+import it.unibo.authsim.library.dsl.cryptography.encrypter.asymmetric.key.{KeyPair, KeysGenerator}
+import it.unibo.authsim.library.dsl.cryptography.util.Base64
+import it.unibo.authsim.library.dsl.cryptography.encrypter.{AsymmetricEncrypter, BasicCipher}
+
+import java.security.*
+import java.security.{KeyPairGenerator, KeyPair as JavaKeyPair}
+import java.security.spec.{PKCS8EncodedKeySpec, X509EncodedKeySpec}
+import javax.crypto.Cipher
+
+/**
+ * RSA cipher object
+ */
+object RSACipher:
+  import it.unibo.authsim.library.dsl.cryptography.util.ImplicitConversion._
+
+  /**
+   * Apply method for the object
+   * @return        an istance of the RSA class
+   */
+  def apply() = new RSAEncrypterImpl()
+
+  /**
+   * Basic implementation of an encrypter which use RSA algorithm for the cryptographic operation
+   */
+  case class RSAEncrypterImpl() extends BasicCipher with AsymmetricEncrypter:
```

Comment:
If the class is just used to provide a basic implementation of the trait, it should be made private:
```suggestion
  private case class RSAEncrypterImpl() extends BasicCipher with AsymmetricEncrypter:
```

Reply:
If I do so that I cannot use the class implementation in the cipher;  my ciphers needs an implementation of the algorithm trait to work, therefore I can:

- put everything in the trait of the algorithm ( not sure it is the right solution)
- not use the private attribute-> encapsulation goes to ....

What do you think, have you thought about other possible solutions, or suggestions?